applications/tari_base_node/src/parser.rs

```diff
@@ -1140,14 +1140,24 @@ impl Parser {
             let mut missing_blocks = Vec::new();
             let mut missing_headers = Vec::new();
             print!("Searching for height: ");
+            // We need to check every header, but not every block.
             while height > 0 {
                 print!("{}", height);
                 io::stdout().flush().unwrap();
-                let block = node.get_blocks(vec![height]).await;
-                if block.is_err() {
-                    // for some apparent reason this block is missing, means we have to ask for it again
-                    missing_blocks.push(height);
-                };
+                // we can only check till the pruning horizon, 0 is archive node so it needs to check every block.
+                if height > meta.pruning_horizon {
```

Comment:
The pruning_horizon should be used to calculate the horizon height using the ChainMetadata horizon_block function. The height should then be larger than the horizon height and not the pruning horizon.

Reply:
Your correct, fixing it now